### ISSUE TITLE
Document running behind proxies

### ISSUE THREAD

**Issue by shabda** [bernardopires/django-tenant-schemas#316]
While it is not directly related to tenant_schemas, documenting how to get this working behind a proxy (nginx or Apache) would be useful. I am glad to write a PR if you are interested to merge this.

**Comment by vinitkumar** [comment]
+1

**Comment by goodtune** [comment]
I'm not sure there is enough to warrant writing custom documentation - there are too many use cases, but it might be worth referencing the relevant sections of the nginx and/or Apache documentation on how to configure a reverse proxy. 

Using variables derived from the hostname could be a good example - use the variable to serve the relevant static media from the filesystem?

**Comment by priyadhoundiyal** [comment]
HI! I'd like some help with this.
I'm using a subfolder structure for selecting the tenants instead of subdomains after modifying [BaseTenantMiddleware](https://github.com/bernardopires/django-tenant-schemas/blob/master/tenant_schemas/middleware.py) by using get_full_path() instead of get_host()
 
However, the requests for my media files will look something like:
http://xyz.com/media/member_photos/my_photo.jpg

How do I then figure out from the url which tenant's media directory I should be looking into?

**Comment by vinitkumar** [comment]
@priyadhoundiyal This is something that I used when I was using this plugin

```python
from django.db import connection
from django.utils.deconstruct import deconstructible

@deconstructible
class UploadPath(object):

    def __init__(self, sub_path):
        self.path = sub_path

    def __call__(self, instance, filename):
        return "genericbanner/%s/%s/%s" % (instance.generic_title, connection.schema_name, filename)

get_upload_path = UploadPath('GenericModelPlugin')

class GenericModel(models.Model):
    generic_image = models.ImageField(verbose_name=_(u'Afbeelding'), help_text=_(u'Image'), upload_to=get_upload_path)
```
Now, as you see the images will be namespaced using the schema_name and it will be saved in the model. So if you use a media source like S3 or something else.

The URL would be something like this:

```
http://api.s3.aws.com/media/tenant1/media.png
```

So, my suggestion is that don't bother finding out the URL, rather just save the info namespaced with tenant name in the database itself.

Hope this helps!

**Comment by priyadhoundiyal** [comment]
Thank you @vinitkumar! :)
I'm currently working with something like this, including tenant.domain_url from connection information in the upload location for images.

```
def get_photo_path(instance, filename):
  date = datetime.now().date()
  upload_folder = '{0}/photos/{1}/{2}'.format(
                             connection.tenant.domain_url,
                             date,
                             filename)
  return upload_folder
```

I couldn't get this to work with TenantFileSystemStorage though because that would create another directory named as the domain_url. 
Just so I'm clear, this means that I don't really have to use TenantFileSystemStorage?
Thanks again!

**Comment by vinitkumar** [comment]
@priyadhoundiyal  I don't think you need to bother about the tenant's domain URL at all. For us, the media storage backend was AWS S3. You can use whatever you like, so if our uploaded media was namespaced properly and saved in the model field then getting the URL is just this:

in settings.py
```
# AWS related settings for production
AWS_STORAGE_BUCKET_NAME = 'assets.dummy.com'
MEDIA_URL = 'http://assets.dummy.com.s3-website-eu-west-1.amazonaws.com/'

DEFAULT_FILE_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
THUMBNAIL_DEFAULT_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
```

```python
image_url = instance.generic_image.url

#which would give:
# http://assets.dummy.com.s3-website-eu-west-1.amazonaws.com//media/tenant1/media.png
```
